mcf-browser/src/TimeGraph.tsx

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {Component} from 'react';
+import {Line} from 'react-chartjs-2';
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+} from 'chart.js';
+
+import {Series} from './back-end/time-series';
+import {LocationMapping} from './TimelineExplorer';
+
+const COLORS = [
+  '#bcf60c',
+  '#fabebe',
+  '#008080',
+  '#e6beff',
+  '#9a6324',
+  '#fffac8',
+  '#800000',
+  '#aaffc3',
+  '#e6194b',
+  '#3cb44b',
+  '#ffe119',
+  '#4363d8',
+  '#f58231',
+  '#911eb4',
+  '#46f0f0',
+  '#f032e6',
+  '#808000',
+  '#ffd8b1',
+  '#000075',
+  '#808080',
+  '#d37295',
+  '#000000',
+];
+
+const LINE_OPTIONS = {
+  interaction: {
+    intersect: false,
+  },
+  scales: {
+    x: {
+      grid: {
+        display: false,
+      },
+    },
+  },
+  plugins: {
+    legend: {
+      display: true,
+      labels: {
+        font: {
+          size: 12,
+        },
+      },
+    },
+  },
+  title: {},
+};
+
+ChartJS.register(
+    CategoryScale,
+    LinearScale,
+    PointElement,
+    LineElement,
+    Title,
+    Tooltip,
+    Legend,
+);
+
+interface LineOptions {
+  [element: string]: Object;
+}
+
+interface DataSet {
+  data: number[];
+  label: string;
+  fill: boolean;
+  borderColor: string;
+}
+
+interface LineData {
+  labels: string[];
+  datasets: DataSet[];
+}
+
+interface TimeGraphPropType {
+  /**
+  * Passes the data to be plotted
+  */
+  data: Series[];
+
+  /**
+  * The title of the graph
+  */
+  title: string;
+
+  /** A mapping from a location dcid to its name */
+  locationMapping: LocationMapping;
+}
+
+interface TimeGraphStateType {
+  /** The lineData for the Line graph component */
+  lineData: LineData;
+
+  /** The lineOptions for the Line graph component */
+  lineOptions: LineOptions;
+}
+
+/** Component to display a single graph */
+class TimeGraph extends Component<TimeGraphPropType, TimeGraphStateType> {
+  /** Constructor for class, sets initial state
+    * @param {TimeGraphPropType} props the props passed in by parent component
```

Comment:
these indentations should be 
```
/**
 *
 */
```

and elsewhere too

Reply:
Fixed!